modules/Bio/EnsEMBL/DBSQL/SliceAdaptor.pm

```diff
@@ -2727,5 +2727,89 @@ sub _build_circular_slice_cache {
   $sth->finish();
 } ## end _build_circular_slice_cache
 
+=head2 create_chromosome_alias
+
+  Args       : none
+  Example    : $self->create_chromosome_alias();
+  Description: create chromosome alias in coordinate system with karyotype attributes
+  Returntype : Bio::EnsEMBL::CoordSystem, or none 
+  Exceptions : None
+  Caller     : general
+  Status     : in testing
+
+=cut
+
+sub _create_chromosome_alias {
+  my $self = shift;
+  my $csa  = $self->db->get_CoordSystemAdaptor();
+
+  print " DEBUG: Noticed you have requested to get a chromosome slice, when no chromosome CoordSystem exists. Finding CoordSystem to alias to...\n";
+
+  # to store reformatted db query results
+  my %karyotype_seq_regions;
+  
+  my $karyotype_rank_string = "karyotype_rank";
+
+  # to store coord_system_id to alias 
+  my $coord_system_id;
+
+  # check whether seq region cache exists, otherwise run database query
+  if ( $self->{'karyotype_cache'} ) {
+    print "Seq region cache exists...\n";
```

Comment:
Do we need this message? How many times will it be displayed in an average script?